src/dtemplate.d

```diff
@@ -4059,27 +4062,46 @@ extern (C++) MATCH deduceType(RootObject o, Scope* sc, Type tparam, TemplatePara
             {
                 result = m;
             }
+            else if (e.type.toBasetype().isintegral() ||
```

Comment:
Yes, also for `Tsarray` and `Tstruct`, we have to check top-qualifier conversion. Will do it later.